qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py

```diff
@@ -14,44 +14,87 @@
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.opflow import Z2Symmetries, commutator, double_commutator, PauliSumOp
 from qiskit.tools import parallel_map
 from qiskit.tools.events import TextProgressBar
 from qiskit.utils import algorithm_globals
-from qiskit.algorithms import AlgorithmResult
-from qiskit.opflow import (
-    Z2Symmetries,
-    commutator,
-    double_commutator,
-    PauliSumOp,
-)
+
+from qiskit.quantum_info import SparsePauliOp
 from qiskit.primitives import BaseEstimator
 
+from qiskit_nature.second_q.algorithms.ground_state_solvers import GroundStateSolver
+from qiskit_nature.second_q.algorithms.ground_state_solvers.ground_state_solver import QubitOperator
+from qiskit_nature.second_q.algorithms.ground_state_solvers.minimum_eigensolver_factories import (
+    MinimumEigensolverFactory,
+)
+from qiskit_nature.second_q.algorithms.excited_states_solvers.excited_states_solver import (
+    ExcitedStatesSolver,
+)
+from qiskit_nature.second_q.mappers import QubitConverter
 from qiskit_nature.second_q.operators import SparseLabelOp
 from qiskit_nature.second_q.problems import (
     BaseProblem,
     ElectronicStructureProblem,
     VibrationalStructureProblem,
+    EigenstateResult,
+    ElectronicStructureResult,
 )
-from qiskit_nature.second_q.problems import EigenstateResult
 
 from .qeom_electronic_ops_builder import build_electronic_ops
 from .qeom_vibrational_ops_builder import build_vibrational_ops
-from .excited_states_solver import ExcitedStatesSolver
-from ..ground_state_solvers import GroundStateSolver
+
 
 logger = logging.getLogger(__name__)
 
 
+class EvaluationRule(Enum):
+    """An enumeration of the available evaluation rules for the excited states solvers.
+
+    This ``Enum`` simply names the available evaluation rules.
+    """
+
+    ALL = "all"
+    DIAG = "diag"
+
+
 class QEOM(ExcitedStatesSolver):
-    """The calculation of excited states via the qEOM algorithm."""
+    """The calculation of excited states via the qEOM algorithm.
+
+    This algorithm approximates the excited-state properties of a problem using additional measurements
+    on the ground state provided by a ``GroundStateSolver`` object.
+    The precision of the ``GroundStateSolver.solve`` method for the ground state approximate directly
+    affects the precision of the qEOM algorithm for the same problem.
+    The ``excitations`` are used to build a linear subspace in which an eigenvalue problem for the
+    projected Hamiltonian will be solved. This method typically works well for calculating the
+    lowest-lying excited states of a problem.
+    The excited-state energies are calculated by default in this algorithm for all excited states.
+    Auxiliary observables can be specified to the ``solve`` method along with auxiliary evaluation
+    rules of the ``QEOM`` object.
+
+    The following attributes can be read and updated once the ``QEOM`` object has been
+    constructed.
+
+    Attributes:
```

Comment:
I don't really know how to deal with it: excitation is not a public attribute in the sense of aux_eval_rules for example. A setter and getter for excitation were defined and because the setter does an additional check, it is used in the init (self.excitations = excitations)
But if I add a docstring for excitations as an attribute, then it creates a conflict with the getter/setter doctoring

Reply:
Hm I actually overlooked that setter, sorry..

Looking at it now though I am a bit confused. The setter retricts the value a lot more than what the `__init__` suggests is possible. Thus, I would argue that this check is in fact obsolete and ought to be outright removed.
With that check removed, you can indeed make this simply a public attribute.

Could you test this quickly by using for example numbers as an input? As things are now I suspect them to break on that check although they should work..

Reply:
As discussed offline, this test is only required to defend against using triple excitations (ok for UCC but not tested for QEOM)